### ISSUE TITLE
no chance to setup listener to "ready" event on HighLevelProducer when client is already "ready"

### ISSUE THREAD

**Issue by royaltm** [SOHU-Co/kafka-node#219]
the title says it all, but for completeness consider this example:

```javascript
var client = new kafka.Client('path-to-kafka')
client.on('ready', function() {
  var producer = new kafka.HightLevelProducer(client);
  producer.on('ready', function() {
    console.log('never fired');
  });
});
```

https://github.com/SOHU-Co/kafka-node/blob/master/lib/highLevelProducer.js#L53

ready event is emitted immediately while initializing producer

i think the correct way would be to postpone connect(); with setImmediate in the producer initializer function
this way clients would have been given opportunity to setup up listeners before they are fired

**Comment by royaltm** [comment]
the same applies to Producer

**Comment by daluu** [comment]
So does this mean we should or need to explicitly register an event listener for client on ready, at least for producing? I didn't see that in the examples in the readme of this site. It just showed producer on ready.

**Comment by daluu** [comment]
Is this also related to #207?

**Comment by royaltm** [comment]
Yes, you should register according to docs, code and #207. To workaround present issue use separate client - one per each producer/consumer and initialize producer/consumer together with client.
